Utest/juelich_tests/test_10/master_ini.xml

```diff
@@ -3,7 +3,7 @@
 <JuPedSim project="JPS-Project" version="0.8" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="ini.xsd">
 
   <!-- seed used for initialising random generator -->
-  <seed>linspace(1, 10000, 10)</seed>
+  <seed>np.linspace(1, 10000, 10)</seed>
```

Comment:
This parameter input option must be supplemented in the [documentation](http://www.jupedsim.org/jpscore/2016-11-01-inifile). It is missing completely so far.

Reply:
> This parameter input option must be supplemented in the [documentation](http://www.jupedsim.org/jpscore/2016-11-01-inifile). It is missing completely so far.

I don't think this is a feature of `jpscore`, in test_10 `makeini.py` is using the `master_ini.xml` as template to generate a couple of ini files.
e.g:
```xml
<JuPedSim xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" project="JPS-Project" version="0.8" xsi:noNamespaceSchemaLocation="ini.xsd">

  
  <seed>1.0</seed>
  <num_threads>4</num_threads>
  <max_sim_time unit="sec">900</max_sim_time>
  
  <geometry>../geometry.xml</geometry>
  
  <trajectories embed_mesh="false" format="xml-plain" fps="8">
    <file location="../trajectories/traj_seed_1.0_linkedcells-cell_size_2.0.xml" />
  </trajectories>
  
  <logfile>log</logfile>

  
  <traffic_constraints>
    <rooms>
      <room room_id="0" state="good" />
    </rooms>
    <rooms>
      <room room_id="1" state="good" />
    </rooms>
    
    <doors>
      <door caption="" state="open" trans_id="2" />
    </doors>
  </traffic_constraints>
  <routing>
    <goals>
      <goal caption="goal 1" final="true" id="0">
        <polygon>
          <vertex px="70" py="101" />
          <vertex px="70" py="103" />
          <vertex px="75" py="103" />
          <vertex px="75" py="101" />
          <vertex px="70" py="101" />
        </polygon>
      </goal>
    </goals>
  </routing>

  
  <agents operational_model_id="3">
    <agents_distribution>
      <group agent_parameter_id="0" goal_id="0" group_id="2" number="50" room_id="1" route_id="" router_id="1" subroom_id="0" />
      
      
    </agents_distribution>

    
    <agents_sources />
  </agents>

  
  <operational_models>
    <model description="Tordeux2015" operational_model_id="3">
      <model_parameters>
        <solver>euler</solver>
        <stepsize>0.05</stepsize>
        <exit_crossing_strategy>3</exit_crossing_strategy>
        <linkedcells cell_size="2.0" enabled="true" />
        <force_ped D="0.1" a="5" />
        <force_wall D="0.02" a="5" />
      </model_parameters>
      <agent_parameters agent_parameter_id="0">
        <v0 mu="1.34" sigma="0.001" />
        <bmax mu="0.15" sigma="0.00000" /> 
        <bmin mu="0.15" sigma="0.00000" />
        <amin mu="0.15" sigma="0.00000" />
        <tau mu="0.5" sigma="0.001" />
        <atau mu="0.0" sigma="0.00000" />
        <T mu="1" sigma="0.001" />
      </agent_parameters>
    </model>

    <model description="gcfm" operational_model_id="1">
      <model_parameters>
        <solver>euler</solver>
        <stepsize>0.001</stepsize>
        <exit_crossing_strategy>4</exit_crossing_strategy>
        <linkedcells cell_size="2.0" enabled="true" />
        <force_ped dist_max="3" disteff_max="2" interpolation_width="0.1" nu="0.2" />
        <force_wall dist_max="3" disteff_max="2" interpolation_width="0.1" nu="0.3" />
      </model_parameters>
      <agent_parameters agent_parameter_id="0">
        <v0 mu="1.34" sigma="0.001" />
        <bmax mu="0.25" sigma="0.001" />
        <bmin mu="0.20" sigma="0.001" />
        <amin mu="0.18" sigma="0.001" />
        <tau mu="0.5" sigma="0.001" />
        <atau mu="0.23" sigma="0.001" />
      </agent_parameters>
    </model>

    <model description="krausz" operational_model_id="5">
      <model_parameters>
        <solver>euler</solver>
        <stepsize>0.001</stepsize>
        <exit_crossing_strategy>4</exit_crossing_strategy>
        <linkedcells cell_size="2.0" enabled="true" />
        <force_ped dist_max="3" disteff_max="2" interpolation_width="0.1" nu="0.2" />
        <force_wall dist_max="3" disteff_max="2" interpolation_width="0.1" nu="0.3" />
      </model_parameters>
      <agent_parameters agent_parameter_id="0">
        <v0 mu="1.34" sigma="0.001" />
        <bmax mu="0.25" sigma="0.001" />
        <bmin mu="0.20" sigma="0.001" />
        <amin mu="0.18" sigma="0.001" />
        <tau mu="0.5" sigma="0.001" />
        <atau mu="0.23" sigma="0.001" />
      </agent_parameters>
    </model>

    <model description="gompertz" operational_model_id="2">
      <model_parameters>
        <solver>euler</solver>
        <stepsize>0.01</stepsize>
        <exit_crossing_strategy>3</exit_crossing_strategy>
        <linkedcells cell_size="2.0" enabled="true" />
        <force_ped b="0.25" c="3.0" nu="3" />
        <force_wall b="0.70" c="3.0" nu="10" />
      </model_parameters>
      <agent_parameters agent_parameter_id="0">
        <v0 mu="1" sigma="0.0" />
        <bmax mu="0.25" sigma="0.001" />
        <bmin mu="0.20" sigma="0.001" />
        <amin mu="0.18" sigma="0.001" />
        <tau mu="0.5" sigma="0.001" />
        <atau mu="0.5" sigma="0.001" />
      </agent_parameters>
    </model>

  </operational_models>

  <route_choice_models>
    <router description="global_shortest" router_id="1">
      <parameters>
        
      </parameters>
    </router>
  </route_choice_models>

</JuPedSim>
```

Reply:
> I don't think this is a feature of jpscore, in test_10 makeini.py is using the master_ini.xml as template to generate a couple of ini files.

Okay, that's right. The entire structure of the test cases is confusing for me. In my opinion, this does not change the fact that we at least have to document this _spaghetti code_.

Another question:

``` <seed>1.0</seed>```

Is a floating point number required to specify the seed value?

Reply:
> Is a floating point number required to specify the seed value?

Actually the IniFileParser **_should_** throw up (ints are expected not floats), but since we are using `atoi` only digits before the `.` get parsed. So basically it behaves like a cast to int :/ (see this for the details of `atoi` https://en.cppreference.com/w/c/string/byte/atoi) 

> Okay, that's right. The entire structure of the test cases is confusing for me. In my opinion, this does not change the fact that we at least have to document this _spaghetti code_.

How about Ill document this in additional readme in `juelich_tests`? This would be rendered in github if you open the folder and is applicable for all test cases in juelich tests.

Reply:
We have to document it. A description in a ReadMe file would be a first step. 

Is there any suggestion or idea to untangle the code?

Reply:
I probably would move away from on-the-fly generation of input and instead write a tool to generate input properly, then commit the generated cases. This would allow for a lot easier debugging in case one of the testcases runs into an error. But this is just very high level for most test cases I have no clue what their test intend is (testcase 10 is a non-descriptive name...)

Reply:
The description of the test cases can be found [here](http://www.jupedsim.org/jpscore/2016-11-02-juelich). We should revise and supplement the documentation in the source (see new issue #466 ).